langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Matches.java

```diff
@@ -53,7 +66,20 @@ public static BArray matchGroupsAt(BRegexpValue regExp, BString str, int startIn
     }
 
     public static boolean isFullMatch(BRegexpValue regExp, BString str) {
-        Matcher matcher = RegexUtil.getMatcher(regExp, str);
+        if (str.length() == 0 || regExp == null) {
+            return false;
+        }
+
+        Matcher matcher = getMatcher(regExp, str);
         return matcher.matches();
     }
+
+    private static Matcher getMatcher(BRegexpValue regExp, BString str) {
+        try {
+            return RegexUtil.getMatcher(regExp, str);
+        } catch (Exception e) {
```

Comment:
Wouldn't catching PatternSyntaxException be enough?

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/39444/commits/4c4adeacdb83b2291c52dff4a09215f5198c305e.